src/Model/Settings.cs

```diff
@@ -219,62 +211,58 @@ public void Load(string filename)
 
         public string FileName
         {
-            get { return this.filename; }
+            get { return this._fileName; }
             set
             {
-                if (this.filename != value)
+                var fileName = value;
+                if (string.IsNullOrWhiteSpace(fileName) && this._fileName != value)
                 {
-                    this.filename = value;
+                    this._fileName = value;
 
                     StopWatchingFileChanges();
 
-                    this.watcher = new FileSystemWatcher(Path.GetDirectoryName(filename),
-                        Path.GetFileName(filename));
+                    this._watcher = new FileSystemWatcher(Path.GetDirectoryName(_fileName), Path.GetFileName(_fileName));
 
-                    this.watcher.Changed += new FileSystemEventHandler(watcher_Changed);
-                    this.watcher.EnableRaisingEvents = true;
+                    this._watcher.Changed += new FileSystemEventHandler(watcher_Changed);
+                    this._watcher.EnableRaisingEvents = true;
                 }
             }
         }
 
         string ConvertToString(object value)
         {
-            if (value is Uri)
-            {
-                return pfn.GetPersistentFileName((Uri)value);
-            }
-            else if (value is string)
-            {
-                return (string)value;
-            }
+            if (value is Uri uri)
+                return _persistentFileNames.GetPersistentFileName(uri);
+            else if (value is string @string)
+                return @string;
```

Comment:
I added some [coding guidlines](https://github.com/microsoft/XmlNotepad/blob/master/docs/dev/index.md), and one of them is that I prefer to keep curly braces on all if-statements, even simple ones like this.  I would also prefer not to use the @ sign to protect identifiers.  So just pick a different variable name from "string".